src/components/views/elements/ServerPicker.tsx

```diff
@@ -70,11 +70,11 @@ const ServerPicker = ({ title, dialogTitle, serverConfig, onServerConfigChange }
     }
 
     let serverName: React.ReactNode = serverConfig.isNameResolvable ? serverConfig.hsName : serverConfig.hsUrl;
-    if (serverConfig.hsNameIsDifferent) {
-        serverName = <TextWithTooltip class="mx_Login_underlinedServerName" tooltip={serverConfig.hsUrl}>
-            { serverConfig.hsName }
-        </TextWithTooltip>;
-    }
+    serverName = <TextWithTooltip
+                    class={serverConfig.hsNameIsDifferent ? "mx_Login_underlinedServerName" : ""}
+                    tooltip={serverConfig.hsUrl}>
+        { serverConfig.hsName }
+    </TextWithTooltip>;
```

Comment:
this no longer really makes sense, L72 sets a string which doesn't get used.
And now there's always a tooltip, even if it is wholly redundant.

Reply:
I wouldn't say wholly redundant. It shows whether http or https is being used, it shows the port being used if HS is using a nonstandard port, it shows the subdomain HS uses for client traffic, etc. Plus it is a tooltip, so it doesn't take up any extra space and only shows up on hover.

<img width="245" src="https://user-images.githubusercontent.com/5855073/148774927-8f8d0710-80ec-4845-92b6-38488a4d276a.png">

<img width="241" src="https://user-images.githubusercontent.com/5855073/148774888-bf9ee055-c601-482e-a805-4b07cfb1e905.png">

Reply:
I believe it was a design/product thing to only show it when the name doesn't match, but I agree with your evaluation

Reply:
Ah but that only happens because of your change, previously it'd show `https://localhost:8700` inline due to the ternary statement you removed

Reply:
It'd be good if this PR, labelled `Truncate long server names on login/register screen` did just that, rather than also tweaking the behaviour of what name is shown when

Reply:
> Ah but that only happens because of your change, previously it'd show https://localhost:8700 inline due to the ternary statement you removed

I'm not sure what you mean. This is what it looks like on develop. No inline additional information and no tooltip with extra information.

<img width="298" src="https://user-images.githubusercontent.com/5855073/148777558-7ad6c31c-4465-4281-9f0b-a3064aedd902.png">

> It'd be good if this PR, labelled Truncate long server names on login/register screen did just that, rather than also tweaking the behaviour of what name is shown when

Well I think this is the correct solution. I'm happy to rename the PR whatever you would like.

Reply:
> I'm not sure what you mean. This is what it looks like on develop. No inline additional information and no tooltip with extra information.

Sounds like `isNameResolvable` may be incorrectly set true for some reason in your case then

Reply:
Ah yes I do see [that](https://github.com/matrix-org/matrix-react-sdk/pull/7498#discussion_r781238926) when I click edit and enter https://localhost:8700 but it doesn't work when https://localhost:8700 is configured as the default (using `default_server_config`).

Reply:
Yeah that's a known behaviour for `default_server_config` - its always trusted to be correct and treated as if it was fetched from .well-known